src/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py

```diff
@@ -42,33 +42,58 @@ def compile_file(
 def compile_str(
     qasm_circuit: str, apply_litinski_transform: bool = True
 ) -> Tuple[List[GUISlice], str]:
-    """Returns gui slices and the text of the circuit as processed in various stages"""
+    """Returns gui slices and string JSON of compilation text)"""
     composer_class = lscc.LatticeSurgeryComputation
     layout_types = lscc.LayoutType
 
-    input_circuit = segmented_qasm_parser.parse_str(qasm_circuit)
-
-    compilation_text = "Input Circuit:\n"
-
-    compilation_text += qkvis.circuit_drawer(
+    input_circuit = qkvis.circuit_drawer(
         qkcirc.QuantumCircuit.from_qasm_str(qasm_circuit)
     ).single_string()
 
-    compilation_text += "\nCircuit as Pauli rotations:\n"
-    compilation_text += input_circuit.render_ascii()
+    parsed_circuit = segmented_qasm_parser.parse_str(qasm_circuit)
+    circuit_as_pauli_rotations = parsed_circuit.render_ascii()
 
     # TODO add user flag
-    input_circuit = input_circuit.get_y_free_equivalent()
+    parsed_circuit = parsed_circuit.get_y_free_equivalent()
 
+    circuit_after_litinski_transform = ""
     if apply_litinski_transform:
-        input_circuit.apply_transformation()
-        input_circuit = input_circuit.get_y_free_equivalent()
-        compilation_text += "\nCircuit after the Litinski Transform:\n"
-        compilation_text += input_circuit.render_ascii()
+        parsed_circuit.apply_transformation()
+        parsed_circuit = parsed_circuit.get_y_free_equivalent()
```

Comment:
Sorry for commenting this on your PR, but I was just going over this code with Alexandru today, a good name here would be to call this pauli_circuit or something like that